garage/tf/samplers/multienv_vectorized_sampler.py

```diff
@@ -0,0 +1,169 @@
+"""On policy sampler for a list of environments."""
+import itertools
+import pickle
+
+import numpy as np
+
+from garage.misc import tensor_utils
+import garage.misc.logger as logger
+from garage.misc.overrides import overrides
+from garage.misc.prog_bar_counter import ProgBarCounter
+from garage.tf.envs import VecEnvExecutor
+from garage.tf.samplers.on_policy_vectorized_sampler import (
+    OnPolicyVectorizedSampler)
+
+
+class MultiEnvVectorizedSampler(OnPolicyVectorizedSampler):
+    """
+    Multi-Environment Vectorized Sampler.
+
+    This sampler is just a multi-envrionment version
+    of OnPolicyVectorizedSampler. It takes a list of
+    different environment and sample from them in the
+    same way as an OnPolicyVectorizedSampler. This is
+    used for meta RL algorithms which need to sample
+    from a set of MDP's.
+
+    Args:
+        algo: An meta RL algorithm.
+        envs: A list of environments.
+        n_envs: The number of environments to be created
+            for each VecEnvExecutor.
+    """
+
+    def __init__(self, algo, envs, n_envs=1):
+        super().__init__(algo=algo, n_envs=n_envs)
+        self.envs = envs
+        self.vec_envs = []
+
+    @overrides
+    def start_worker(self):
+        """Create a list of vectorized executors."""
+        n_envs = self.n_envs
+        if getattr(self.algo.env, 'vectorized', False):
+            for env in self.envs:
+                self.vec_envs.append(
+                    self.algo.env.vec_env_executor(
+                        n_envs=n_envs,
+                        max_path_length=self.algo.max_path_length))
+        else:
+            for env in self.envs:
+                envs = [pickle.loads(pickle.dumps(env)) for _ in range(n_envs)]
+                self.vec_envs.append(
+                    VecEnvExecutor(
+                        envs=envs, max_path_length=self.algo.max_path_length))
+        self.env_spec = self.envs[0].spec
+
+    @overrides
+    def obtain_samples(self, itr, batch_size=None):
+        """
+        Sample from environments.
+
+        Args:
+            itr: Iteration number.
+            batch_size: The number of samples.
+        """
+        logger.log("Obtaining samples for iteration %d..." % itr)
+        all_paths = []
+
+        if not batch_size:
+            batch_size = self.algo.max_path_length * self.n_envs * len(
+                self.envs)
+        batch_size_per_task = self.algo.max_path_length * self.n_envs
```

Comment:
Should we make `batch_size_per_task` configurable?

Also, since `batch_size` is used as an indicator for the progress bar, and has nothing to do with the actual number of samples, maybe just replace `batch_size` with `batch_size_per_task` in the interface.